### PR TITLE
Fix audio streaming via BaseProgressiveStreamingService

### PR DESCRIPTION
**Pull request by WillWill56**
**Changes**
Added a httpClient argument to the constructors of relevant classes to allow the BaseProgressiveStreamingService class to acquire an instance. May have added it to a few too many classes, could remove it from BaseStreamingService and other classes that are derived from it if necessary.

**Issues**
This resolves one of the problems I was having with jellyfin-plugin-tunein, and potentially other audio streaming issues. (see https://github.com/jellyfin/jellyfin-plugin-tunein/pull/2 )

### CODE REVIEW

**Comment by EraYaN** [comment]
Is this HttpClient just never used? At least not in this set of changes.

**Comment by WillWill56** [comment]
There's a single use of HttpClient in GetStaticRemoteStreamResult in the file BaseProgressiveStreamingService.cs, looks like it gets the content type maybe? I'm not 100% sure, and I'll admit this is a bit of a "quick and dirty" fix, it's been sitting there since commit 1d7d52f which looks to be the original port of all this code over to JellyFin. Perhaps there is another solution that I'm not seeing, I was just trying to make as few changes as possible.

**Comment by JustAMan** [comment]
The only reference I see there is using HttpClient to download fonts, thing that IIRC we removed.
Where do you want to use this HttpClient?

**Comment by WillWill56** [comment]
I don't want to use the HttpClient anywhere? There was already a HttpClient in the base class "BaseStreamingService", and it already had one reference in BaseProgressiveStreamingService.cs on line 303, but since it was never assigned a value, it was always a null reference. The call to HttpClient.GetResponse() would throw a null exception when the server tries to open a remote audio stream (e.g. internet radio). This was all in the code before my commits.

I see nothing about fonts in the code I'm referring to either?

**Comment by JustAMan** [comment]
> and it already had one reference in BaseProgressiveStreamingService.cs on line 303, but since it was never assigned a value, it was always a null reference

Now _that_ is explaining things!
Would be much clearer if you just indicated a stacktrace with `NullPointerException` and say "I fixed this" IMHO.

**Comment by WillWill56** [comment]
... there's a stack trace in my comment in the pull request I linked?

**Comment by JustAMan** [comment]
Oh, please accept my apologies then, I didn't look in there :(

**Comment by WillWill56** [comment]
It's fine, I should've made it clearer or just copied it here, I don't have much experience with github so I need to learn to deal with stuff like this haha.

File: MediaBrowser.Api/Playback/BaseStreamingService.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ public abstract class BaseStreamingService : BaseApiService
         protected IMediaSourceManager MediaSourceManager { get; private set; }
         protected IJsonSerializer JsonSerializer { get; private set; }
 
-        public static IHttpClient HttpClient;
+        public static IHttpClient HttpClient { get; private set; }
```
Comment (cvium) [inline MediaBrowser.Api/Playback/BaseStreamingService.cs]: Doesn't really make sense as a static anymore

Comment (WillWill56) [reply MediaBrowser.Api/Playback/BaseStreamingService.cs]: Ah, good catch, should make it protected from what I can tell. Maybe I should move it into the BaseProgressiveStreamingService class? Would reduce the number of classes that need a HttpClient in their constructor.

**Comment by WillWill56** [comment]
If you're interested, I'm testing with a tunein plugin compiled from this specific commit: https://github.com/Deanosim/jellyfin-plugin-tunein/commit/159d19248a54b8aa95f4ab41a1c2596fe86af195 . Some radio stations are actually playing now, the plugin needs a lot more work but it's nice to see some basic functionality.

**Pull request merged**